src/UI/Art/SpriteArtist/PlayerSetupAiArtist.java

```diff
@@ -0,0 +1,111 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import AI.AILibrary;
+import AI.AIMaker;
+import Engine.IController;
+import UI.PlayerSetupAiController;
+
+public class PlayerSetupAiArtist
+{
+  private static HashMap<Integer, BufferedImage> aiNameplates = new HashMap<Integer, BufferedImage>();
+  private static int maxAiNameLength = -99;
+
+  public static void draw(Graphics g, IController controller, Color playerColor)
+  {
+    PlayerSetupAiController control = (PlayerSetupAiController)controller;
+    if( null == control )
+    {
+      System.out.println("WARNING! PlayerSetupAiController was given the wrong controller!");
+    }
+
+    // Define the draw space
+    int drawScale = SpriteOptions.getDrawScale();
+    Dimension dimensions = SpriteOptions.getScreenDimensions();
+    int myWidth = dimensions.width / drawScale;
+    int myHeight = dimensions.height / drawScale;
+    BufferedImage image = SpriteLibrary.createTransparentSprite(myWidth, myHeight);
+    Graphics myG = image.getGraphics();
+
+    /////////////// AI Names //////////////////////
+    ArrayList<AIMaker> aiOptions = AILibrary.getAIList();
+    if(maxAiNameLength < 0) // Initialize our name tags if we haven't yet.
+    {
+      // Figure out the longest-length name.
+      for( int i = 0; i < aiOptions.size(); ++i )
+      {
+        if( aiOptions.get(i).getName().length() > maxAiNameLength ) maxAiNameLength = aiOptions.get(i).getName().length();
+      }
+
+      // Make a string of all spaces of the required length so we can build nameplates of the same length. Soooo hacky, I know.
+      StringBuilder buf = new StringBuilder();
+      for( int n = 0; n < maxAiNameLength; ++n ) buf.append(" ");
```

Comment:
Why do you say this is hacky? You could do the same by messing with the hbuffer (which would center the names), but this is probably a lot easier and arguably as good or better.